galley/pkg/runtime/projections/serviceentry/converter/instance.go

```diff
@@ -65,14 +65,19 @@ func (i *Instance) convertService(service *resource.Entry, outMeta *mcp.Metadata
 	}
 
 	spec := service.Item.(*coreV1.ServiceSpec)
-
-	resolution := networking.ServiceEntry_STATIC
 	location := networking.ServiceEntry_MESH_INTERNAL
 	endpoints := convertExternalServiceEndpoints(spec, service.Metadata)
+	var resolution networking.ServiceEntry_Resolution
+	// Resolution STATIC must have endpoints
+	if len(endpoints) != 0 {
+		resolution = networking.ServiceEntry_STATIC
```

Comment:
This isn't quite right ... the endpoints created above are for external services only. 

Pilot will fail validation for STATIC with no endpoints? Maybe we have to pass the existing endpoints into this function as well?

Reply:
Yeah pilot has specific validation for STATIC with no endpoints (here)[https://github.com/istio/istio/blob/master/pilot/pkg/model/validation.go#L2377]. I will look into passing the existing endpoints.